# Linkage-drag

   Pipeline for SNP calling can be found in our previous project [here](https://github.com/owensgl/haploblocks/tree/master/variantcall)
   
   Pipeline for gene annotation filtration can be found [here](https://github.com/megahitokiri/Sunflower_annotation_Snakemake)
   
* [Variants](https://github.com/hkchi/Linkage-drag/tree/master/Variants)
   
   Scripts for identifying sequence and structural variants between genome assemblies
   
* [Introgression](https://github.com/hkchi/Linkage-drag/tree/master/Introgression)
   
   Scripts for identifying introgressed regions in the genomes

* [SVs](https://github.com/hkchi/Linkage-drag/tree/master/SVs)
   
   Scripts for genetic variation analyses

* [Deleterious](https://github.com/hkchi/Linkage-drag/tree/master/Deleterious)
   
   Scripts for analyzing deleterious load
   
* [PAV](https://github.com/hkchi/Linkage-drag/tree/master/PAV)
   
   Scripts for analyzing gene PAVs